fenicsadapter/fenicsadapter.py

```diff
@@ -55,54 +64,98 @@ def eval(self, value, x):
+    :ivar _write_data_id: ID of the data on the coupling mesh created from data name
+    :ivar _write_data: actual data to write
+    :ivar _read_data_name: name of read data as defined in preCICE config (from JSON config file)
+    :ivar _read_data_id: ID of the data on the coupling mesh created from data name
+    :ivar _read_data: actual data to read
+    :ivar _precice_tau: timestep
+
+    :func convert_fenics_to_precice(): converts FEniCS function, into an array of values
+    :func set_coupling_mesh(): sets the coupling mesh
+    :func extract_coupling_boundary_vertices(): extracts verticies which lay on the boundary
+    :func set_coupling_mesh: sets the coupling mesh
+    :func set_write_field: sets the write field
+    :func set_read_field: sets the read field
+    :func create_coupling_boundary_condition():
+    :func create_coupling_dirichlet_boundary_condition():
+    :func create_coupling_neumann_boundary_condition():
+    :func advance(): advances the simulation
+    :func initialize(): initializes remaining attributes
+    :func is_coupling_ongoing(): determines whether simulation should continue
+    :func extract_coupling_boundary_coordinates(): extracts the coordinates of boundary vertices
+    :func finalize(): finalizes the coupling interface.
+    """
     def __init__(self):
-        self._solver_name = None  # name of the solver, will be configured later
+        self._config = Config()
 
-        self._interface = None  # coupling interface, will be initialized later
+        self._solver_name = self._config._solver_name
 
-        self._dimensions = None
+        self._interface = PySolverInterface.PySolverInterface(self._solver_name, 0, 1)
+        self._interface.configure(self._config._config_file_name)
+        self._dimensions = self._interface.getDimensions()
 
-        self._coupling_subdomain = None  # FEniCS subdomain defining the coupling interface
-        self._mesh_fenics = None  # FEniCS mesh where the coupled simulation takes place
-        self._coupling_bc_expression = None  # Expression describing the coupling condition
+        self._coupling_subdomain = None # initialized later
+        self._mesh_fenics = None # initialized later
+        self._coupling_bc_expression = None # initialized later
 
-        ## coupling mesh related quantities will be defined later
-        self._coupling_mesh_vertices = None  # mesh vertices in a format that can be understood by preCICE
-        self._mesh_name = None  # name of mesh as defined in preCICE config
-        self._mesh_id = None  # ID of the coupling mesh created from mesh name
-        self._vertex_ids = None  # ID of vertices, will be filled by preCICE
-        self._n_vertices = None  # number of vertices
+        ## coupling mesh related quantities
+        self._coupling_mesh_vertices = None # initialized later
+        self._mesh_name = self._config._coupling_mesh_name
+        self._mesh_id = None # initialized later
```

Comment:
I think it is also possible set `_mesh_id` at this point. Currently, it is set in `set_coupling_mesh`.

Reply:
Agree.